File: codegen/codegen-erlang/src/test/resources/ir/rest_xml_encode_get_name_response.expected.erl
%% @doc Encode REST-XML response for smithy.beam.demo.http#GetName.
-spec encode_get_name_response(get_name_output()) -> #http_response{}.
encode_get_name_response(#get_name_output{name = Name}) ->
    Headers = [{<<"Content-Type">>, <<"application/xml">>}],
    XmlNs = xml_namespace(),
    MemberMap = maps:filter(
        fun(_, V) ->
            V =/= undefined
        end,
        #{<<"Name">> => Name}
    ),
    Body = encode_xml(#{<<"GetNameOutput">> => MemberMap}, XmlNs),
    #http_response{
        status = 200,
        headers = Headers,
        body = Body
    }.

File: codegen/codegen-elixir/src/test/resources/ir/rest_xml_encode_get_name_response.expected.ex
@spec encode_get_name_response(HttpServiceTypes.GetNameOutput.t()) :: map()
def encode_get_name_response(output) do
  headers = [{"Content-Type", "application/xml"}]
  member_map = %{"Name" => output.name}
  body = encode_xml(%{"GetNameOutput" => member_map}, xml_namespace())
  %RuntimeTypes.HttpResponse{
    status: 200,
    headers: headers,
    body: body
  }
end
